structuredheadings/plugin.js

```diff
@@ -295,21 +295,42 @@
         },
 
         onClick: function (value) {
+          var block = CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor);
           if (value === "restart") {
             editor.execCommand("restartNumbering");
           } else if (value === "clear") {
-            clearStyles(editor, editor.elementPath().block);
-            clearLevel(editor, editor.elementPath().block);
-            clearNumbering(editor, editor.elementPath().block);
+            clearStyles(editor, block);
+            clearLevel(editor, block);
+            clearNumbering(editor, block);
           } else {
-            setNumbering(editor, editor.elementPath().block);
-            setLevel(editor, editor.elementPath().block);
-            setCurrentStyle(editor, editor.elementPath().block, value);
+            // convert any non-numbered headings to numbered
+            var selection = editor.getSelection();
+            if (selection.getRanges()[0].collapsed) {
+              this.setAutonumberClassesForHeading(block);
+            } else {
+              var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
+                editor,
+                selection
+              );
+              if (headings.length > 0) {
+                for (var i = 0; i < headings.length; i++) {
+                  this.setAutonumberClassesForHeading(headings[i]);
+                }
+              }
+            }
+            // apply the correct bulletstyle for all numbered headings
             editor.execCommand("reapplyStyle", value);
+            // set the combo box value
             this.setValue(value, value);
           }
         },
 
+        setAutonumberClassesForHeading: function (heading, value) {
```

Comment:
Oops.  Apparently this current style thing doesn't actually matter anymore, but I will fix it